### PR TITLE
Only add _type if ES version < 8

### PR DESCRIPTION
**Pull request by robbavey**

### CODE REVIEW

**Comment by robbavey** [comment]
Symptoms of this issue look like those described in  https://github.com/elastic/logstash/issues/11264 with ingestion not possible and log entries containing:

```
[2019-10-23T20:23:28,933][INFO ][logstash.agent ] Pipelines running {:count=>2, :running_pipelines=>[:".monitoring-logstash", :main], :non_running_pipelines=>[]}
[2019-10-23T20:23:29,466][ERROR][logstash.outputs.elasticsearch][main] Encountered a retryable error. Will Retry with exponential backoff {:code=>400, :url=>"https://10.0.2.15:9200/_bulk"}
```

**Comment by LeeDr** [comment]
Fixes: https://github.com/elastic/logstash/issues/11264  ?

File: lib/logstash/outputs/elasticsearch/common.rb [hunk 1]
```diff
@@ -70,10 +70,12 @@ def event_action_tuple(event)
       params = {
         :_id => @document_id ? event.sprintf(@document_id) : nil,
         :_index => event.sprintf(@index),
-        :_type => get_event_type(event),
         routing_field_name => @routing ? event.sprintf(@routing) : nil
       }
 
+      params[:_type] = get_event_type(event) if client.maximum_seen_major_version < 8
+
```
Comment (colinsurprenant) [inline lib/logstash/outputs/elasticsearch/common.rb]:
```suggestion
```

**Comment by colinsurprenant** [comment]
LGTM - left cosmetic comment.

**Comment by robbavey** [comment]
Thanks @colinsurprenant - I committed your suggestion

**Comment by manzoor77** [comment]
**[2023-03-17T11:25:11,285][ERROR][logstash.outputs.elasticsearch][main][3404e8877b56c05b348412b7a316fdaceb1867306a80dce71719c975094faeb9] Encountered a retryable error (will retry with exponential backoff) {:code=>400, :url=>"http://localhost:9200/_bulk", :content_length=>41360}**

I need to create indexing of exiting postgres database in elasticsearch. For this purpose I have setup elasticsearch 7.17.4, kibana 7.17.4 and logstash 7.17.4 on my local machine. I have downloaded csv file of posts table from db that contain almost 62k rows of data in it. I have setup logstash config file as per mentioned requirements of logstash doc. i.e.
input {
  file {
    path => "/Users/manzoorfaisal/Desktop/Laptop-Migration-2/logstash-7.17.4/source-posts.csv"
   start_position => "beginning"
   sincedb_path => "/dev/null"
  type => "doc"
   #skip_header => true
   #source => "message"

  }
}
filter {
  # Add your filters here
  csv{
    separator=> ","
    columns=>["id","createat","updateat","deleteat","userid","channelid","rootid","originalid","message","type","props","hashtags","filenames","fileids","hasreactions","editat","ispinned","remoteid"]
 
skip_header => true
  }

  mutate {
    remove_field => ["updateat","deleteat", "rootid","originalid","props","filenames","fileids","hasreactions","editat", "ispinned", "remoteid", "@timestamp","@version","host","path"]
  add_field => {
      "teamid" => "new_value"
    }
  }
}

output {
  elasticsearch {
    hosts => ["localhost:9200"]
    index => "post_index-7"
    document_id => "%{id}"
    
  }

  stdout{codec => rubydebug }
}
I need to create index of existing db in elasticsearch using logstash or using any other method. using logstash I am facing above issue. Can anyone get me out from this issue? or suggest any other authentic method for reindexing of existing postgres db tables into elasticsearch index.

**Pull request merged**